examples/plot_conditional_vs_marginal_xor_data.py

```diff
@@ -0,0 +1,143 @@
+    aspect="auto",
+)
+sns.scatterplot(
+    x=X[:, 0],
+    y=X[:, 1],
+    hue=Y,
+    ax=ax,
+    palette="muted",
+)
+ax.axis("off")
+ax.legend(
+    title="Class",
+)
+ax.set_title("Decision function of SVC with RBF kernel")
+plt.show()
+
+
+###############################################################################
+# Computing the conditional and marginal importance
+# -----------------------------------------------------
+# We first compute the marginal importance by fitting univariate models on each feature.
+# Then, we compute the conditional importance using the CPI class. The univarariate
+# models don't perform above chance, since solving the XOR problem requires to use both
+# features. Conditional importance, on the other hand, reveals that both features
+# are important (therefore rejecting the null hypothesis
+# :math:`Y \perp\!\!\!\perp X^1 | X^2`).
+cv = KFold(n_splits=5, shuffle=True, random_state=0)
+clf = SVC(kernel="rbf", random_state=0)
+# Compute marginal importance using univariate models
+marginal_scores = []
+for i in range(X.shape[1]):
+    feat_scores = []
+    for train_index, test_index in cv.split(X):
+        X_train, X_test = X[train_index], X[test_index]
+        y_train, y_test = Y[train_index], Y[test_index]
+
+        X_train_univariate = X_train[:, i].reshape(-1, 1)
+        X_test_univariate = X_test[:, i].reshape(-1, 1)
+
+        univariate_model = clone(clf)
+        univariate_model.fit(X_train_univariate, y_train)
+
+        feat_scores.append(univariate_model.score(X_test_univariate, y_test))
+    marginal_scores.append(feat_scores)
+
+
+importances = []
+for i, (train_index, test_index) in enumerate(cv.split(X)):
+    X_train, X_test = X[train_index], X[test_index]
+    y_train, y_test = Y[train_index], Y[test_index]
+
+    clf_c = clone(clf)
+    clf_c.fit(X_train, y_train)
+
+    vim = CPI(
+        estimator=clf_c,
+        method="decision_function",
+        loss=hinge_loss,
+        imputation_model_continuous=RidgeCV(np.logspace(-3, 3, 10)),
+        n_permutations=50,
+        random_state=0,
+    )
+    vim.fit(X_train, y_train)
+    importances.append(vim.score(X_test, y_test)["importance"])
+
+importances = np.array(importances).T
+
+# Plotting the results
+fig, axes = plt.subplots(1, 2, sharey=True, figsize=(6, 2.5))
+# Marginal scores boxplot
+sns.boxplot(
+    data=np.array(marginal_scores).T,
+    orient="h",
+    ax=axes[0],
+    fill=False,
+    color="C0",
+    linewidth=3,
+)
+axes[0].axvline(x=0.5, color="k", linestyle="--", lw=3)
+axes[0].set_ylabel("")
+axes[0].set_yticks([0, 1], ["X1", "X2"])
+axes[0].set_xlabel("Marginal Scores (accuracy)")
+axes[0].set_ylabel("Features")
+
+# Importances boxplot
+sns.boxplot(
+    data=np.array(importances).T,
+    orient="h",
+    ax=axes[1],
+    fill=False,
+    color="C0",
+    linewidth=3,
+)
+axes[1].set_xlabel("Conditional Importance")
+axes[1].axvline(x=0.0, color="k", linestyle="--", lw=3)
+plt.tight_layout()
+plt.show()
```

Comment:
Can you add an interpretation of the figure?

Reply:
done